org.lflang.ui/src/org/lflang/ui/actions/CompileActionHandler.java

```diff
@@ -134,15 +134,18 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
             var xtextEditor = EditorUtils.getActiveXtextEditor(event);
             if (xtextEditor != null) {
                 var xtextDocument = xtextEditor.getDocument();
+                var resource = xtextEditor.getResource();
                 if (xtextDocument != null) {
                     // Save editor
                     if (xtextEditor.isDirty()) {
                         xtextEditor.doSave(new NullProgressMonitor());
+                    } else if (resource != null) {
```

Comment:
This should not be an `else if` but a separate `ìf` because now you do not add the file to the compilation if you have to save.
The idea is to save first, if dirty, and then add the resource. Otherwise you have to click the compile button twice, fist to save and then a second time to actually compile.